camel-chimera-graph/src/main/java/com/cefriel/graph/RDFGraph.java

```diff
@@ -41,19 +44,24 @@ public void setBaseIRI(IRI baseIRI) {
         this.baseIRI = baseIRI;
     }
 
-    public void setNamedGraph(String namedGraph) {
-        if (namedGraph != null && !namedGraph.equals("")) {
-            setNamedGraph(Utils.stringToIRI(namedGraph));
+    public void setNamedGraphs(String namedGraphs) {
+        if (namedGraphs != null && !namedGraphs.equals("")) {
+            List<IRI> graphs = Arrays.stream(namedGraphs.split(";")).map(Utils::stringToIRI).toList();
+            setNamedGraphs(graphs);
         }
     }
 
-    public void setNamedGraph(IRI namedGraph) {
-        this.namedGraph = namedGraph;
-        if (this.repo != null) {
+    public void setNamedGraphs(List<IRI> namedGraphs) {
+        if (!namedGraphs.isEmpty() && this.repo != null) {
             ContextAwareRepository cRep = new ContextAwareRepository(this.repo);
-            cRep.setReadContexts(namedGraph);
-            cRep.setInsertContext(namedGraph);
+            cRep.setReadContexts(namedGraphs.toArray(new IRI[0]));
+            cRep.setInsertContext(namedGraphs.get(0));
```

Comment:
A ContextAwareRepository can have a single context (namedGraph) either because the user passed one namedGraph or because the namedGraph was generated by us using the Exchange id. To handle this latter case we have to set the insert context so that triples written in a file without specifying a graph can be inserted in the randomly generated namedGraph. This is because the user cannot possibly know in advance the name of this randomly generated graph.

https://github.com/cefriel/chimera/blob/80dc18b7424b20731d2c56d47868745c0e267ba3/camel-chimera-graph/src/main/java/com/cefriel/graph/RDFGraph.java#L54-L65